lib/core/error/error-with-props.ts

```diff
@@ -0,0 +1,33 @@
+import { isPlainObject } from "@lib/core/typeguards";
```

Comment:
Love this, can we move the asStringOrNull here - etc?

Reply:
I believe that If we add `zod`, we can completely remove the `asString` stuff.

Reply:
This is taken straight from lodash which we already include, why not use `_.isPlainObject()`?